### DISCUSSION TITLE
Client API - POST Channel - No library creation

### DISCUSSION THREAD

**Discussion by boukenka** [nextgenhealthcare/connect#5133] [General]
Hello,

I post via the POST URL api/channels an XML containing a library, 
It does not create a library.
And I don't see any option in the URL to force the creation of a library.
When I import via the UI the same XML, the libraries are detected.
In the popup, I have the choice to create the library or not.
Why doesn't the POST/channels api offer to create the library? 
Or is there a hidden option?

Thanks for any answer.

**Comment by pacmano1** [comment]
I think the existing design expects you to maintain code templates and channels via the api as separate things.    I would open a feature request in the "issues" section.

**Reply by boukenka** [reply]
Thank you
